src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs

```diff
@@ -108,11 +142,18 @@ public bool IsMetadataVirtualLocked
                 get { return (_flags & IsMetadataVirtualLockedBit) != 0; }
             }
 
+            public bool IsVarArg
+            {
+                get { return (_flags & IsVarArgBit) != 0; }
+                set { ThreadSafeFlagOperations.Set(ref _flags, value ? IsVarArgBit : 0); }
```

Comment:
> set { ThreadSafeFlagOperations.Set(ref _flags, value ? IsVarArgBit : 0); } [](http://example.com/codeflow?start=15&length=75)

This doesn't look like the right thing to do. If the value is lazy calculated, then there should be a flag indicating that it has been calculated and an attempt to get the value should trigger the calculation. Alternatively, if the value is easy to calculate from the syntax, we should do that and store the final value from the start, eliminating the ability to adjust the value later.

Reply:
Sure.  I can move this to be computed up front.